controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -157,4 +208,28 @@ protected void shutDown() throws Exception {
             LoggerHelpers.traceLeave(log, objectId, "shutDown", traceId);
         }
     }
+
+    @VisibleForTesting
+    static class PendingSweepQueue {
+        private final ConcurrentLinkedQueue<Supplier<CompletableFuture<Void>>> workQueue = new ConcurrentLinkedQueue<>();
+        private final AtomicBoolean isSealed = new AtomicBoolean(false);
+
+        @Synchronized
+        @VisibleForTesting
+        boolean enqueue(Supplier<CompletableFuture<Void>> futureSupplier) {
```

Comment:
there is a race between while sweep is happening and if any events come. 
they may become lost if we follow as you suggest